server/src/main/java/org/elasticsearch/action/bulk/BulkProcessor.java

```diff
@@ -213,20 +216,21 @@ public static Builder builder(Client client, Listener listener) {
     /**
      * @param consumer The consumer that is called to fulfil bulk operations
      * @param listener The BulkProcessor listener that gets called on bulk events
+     * @param name     The name of this processor, e.g. to identify the scheduler threads
      * @return the builder for BulkProcessor
      */
-    public static Builder builder(BiConsumer<BulkRequest, ActionListener<BulkResponse>> consumer, Listener listener) {
+    public static Builder builder(BiConsumer<BulkRequest, ActionListener<BulkResponse>> consumer, Listener listener, String name) {
```

Comment:
since this class is the [documented approach](https://www.elastic.co/guide/en/elasticsearch/client/java-rest/current/java-rest-high-document-bulk.html#java-rest-high-document-bulk-processor) for clients ingestion, can we preserve binary compatibility here ? (i.e. preserve the original method signature and pass an empty/hard-coded string)

Reply:
Acked, yes, I added the original method back and marked it deprecated so we can remove it at a later date, see 92ea717.